app/views/clients/rollup/_demographics.html.haml

```diff
@@ -36,10 +36,12 @@
         %td= @client.gender
         %td= @client.race_description(include_missing_reason: true)
         %td= HudUtility2024.no_yes_reasons_for_missing_data @client.VeteranStatus
-        %td{colspan: 3}
+        %td{colspan: 2}
 
       - source_clients.joins(:data_source).each do |client|
-        = render 'clients/demographic_row', client: client, authoritative_data_source_ids: authoritative_data_source_ids
+        - hmis_hostname = hmis_data_sources[client.data_source_id]
+        - hmis_url = "https://#{hmis_hostname}/client/#{client.id}" if hmis_hostname
```

Comment:
Hmm, yeah, I think we should package up the generation of HMIS urls, this makes me nervous to have them in multiple locations including views.  If we want to make it simple for now, a method in application_helper.rb would be ok (pass in item type and id), but I think better to put that into DataSource (or something like HmisHelper.)

Reply:
Also, if we're using this only in the `clients/demographic_row` partial, let's move this logic in there instead of passing it in.  That will prevent problems if we re-use the partial elsewhere and don't update them to include `hmis_url:`

Reply:
Ok, I thought that we needed to pass the variables in because of performance. For example right now we are passing `authoritative_data_source_ids`, I assume the reason for that is because we don't want to have to look up the data sources once for each row. So, maybe I should just pass in a `hmis_data_sources` var that is `GrdaWarehouse::DataSource.hmis` ? or am I misunderstanding haml?

Reply:
Huh. To me it looks like it is a db call to get the HMIS data sources (and/or figure out if this client is from hmis). But I moved it to the partial and added a `preload` https://github.com/greenriver/hmis-warehouse/pull/3651/commits/51117b5c6b615132f9bf978ce1e1fc01f90807f4, is that what you're looking for? I think we need the preload to avoid a bunch of db queries.

Reply:
Ah, you are correct about the DataSource load, the preload should help.  If this ends up being an issue we can cache the ids of DataSource.hmis and do comparisons with that, but I'm guessing this will be fine given where we expose those links.